app/lib/report_helper.rb

```diff
@@ -32,29 +32,75 @@ def compute_taxon_zscores(report)
     zscore_array
   end
 
-  def taxonomy_details(view_level, report, params)
+  def taxonomy_details(report, params)
     taxon_zscores = compute_taxon_zscores(report)
-    wanted_level = view_level == 'species' ? TaxonCount::TAX_LEVEL_SPECIES : TaxonCount::TAX_LEVEL_GENUS
-    data = taxon_zscores.group_by { |h| h[:tax_level] == wanted_level && h[:hit_type] }
-    nt = data['NT'] || []
-    nr = data['NR'] || []
-    htc = highest_tax_counts(nt, nr)
+    genus_level = TaxonCount::TAX_LEVEL_GENUS
+    species_level = TaxonCount::TAX_LEVEL_SPECIES
+    tax_ids_str = ''
+
+    taxon_zscores.each do |taxon|
+      tax_ids_str += taxon[:tax_id].to_s
+      tax_ids_str += ','
+    end
+    tax_ids_str = tax_ids_str.chomp(',')
+    lineage_arr = TaxonLineage.connection.select_all("SELECT taxid, superkingdom_taxid,  genus_taxid FROM taxon_lineages WHERE taxid IN (#{tax_ids_str})").to_hash
+    lineage_info = lineage_arr.group_by { |h| h['taxid'] }
+
+    category_taxids = ''
+    lineage_info.each do |taxid|
+      category_taxids += taxid[1][0]['superkingdom_taxid'].to_s
```

Comment:
I think this code doesn't do what you intend.

Don't you mean
```
  lineage_info.each do |taxid, taxons|
    category_taxids += taxons[0]['superkindgdom_tax_id'].to_s
```
and may be only if the superkingdom_tax_id is not null?